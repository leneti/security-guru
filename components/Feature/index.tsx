export { default as Feature } from "./Feature";
export * from "./types";
