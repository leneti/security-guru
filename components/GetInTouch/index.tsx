export { default, type ContactForm } from "./GetInTouch";
