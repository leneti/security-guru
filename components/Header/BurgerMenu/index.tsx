export { default as BurgerMenu } from "./BurgerMenu";
export * from "./types";
