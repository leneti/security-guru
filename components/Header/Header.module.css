.smallDisplayOnly {
  display: none;

  @media (max-width: $mantine-breakpoint-xs) {
    display: initial;
  }
}

.bigDisplayOnly {
  display: initial;

  @media (max-width: $mantine-breakpoint-xs) {
    display: none;
  }
}

.outer {
  width: 100%;
  background-color: var(--mantine-color-background);
  position: sticky;
  top: 0;
  z-index: 2;
  height: var(--header-height);
  box-shadow: var(--mantine-shadow-xl);
}

.inner {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: var(--header-height);
  max-width: var(--mantine-breakpoint-xl);
  margin-left: auto;
  margin-right: auto;

  @media (max-width: calc(var(--mantine-spacing-xl) * 2 + var(--mantine-breakpoint-xl))) {
    padding-left: calc(
      var(--mantine-spacing-xl) - ((100vw - var(--mantine-breakpoint-xl)) / 2)
    );
    padding-right: calc(
      var(--mantine-spacing-xl) - ((100vw - var(--mantine-breakpoint-xl)) / 2)
    );
  }

  @media (max-width: $mantine-breakpoint-xl) {
    padding-left: var(--mantine-spacing-xl);
    padding-right: var(--mantine-spacing-xl);
  }
}

.linkGroup {
  display: flex;
}

.link {
  display: block;
  line-height: 1rem;
  padding: 8px 12px;
  border-radius: var(--mantine-radius-sm);
  text-decoration: none;
  color: var(--mantine-color-dark-0);
  font-size: var(--mantine-font-size-sm);
  font-weight: 500;

  &:hover {
    background-color: var(--mantine-color-dark-4);
  }
}

.subLink {
  text-decoration: none;
  color: var(--mantine-color-dark-0);
  font-size: var(--mantine-font-size-sm);
  font-weight: 500;
}

.linkLabel {
  margin-right: 5px;
}

.switch {
  background-color: var(--mantine-color-dark-5);
  color: var(--mantine-primary-color);
  border: none;
  border-radius: var(----mantine-radius-md);

  &:hover {
    background-color: var(--mantine-color-dark-4);
  }
}
