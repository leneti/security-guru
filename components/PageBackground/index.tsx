export { default, type PageBGProps } from "./PageBackground";
