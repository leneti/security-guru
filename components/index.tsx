export { default as ContactInfo } from "./ContactInfo";
export { default as Footer } from "./Footer";
export { default as GetInTouch, type ContactForm } from "./GetInTouch";
export { default as Header } from "./Header";
export { default as Hero } from "./Hero";
export { default as Logo } from "./Logo";
export { default as PageTitle } from "./PageTitle";
export { default as PageBackground, type PageBGProps } from "./PageBackground";
export { default as SlideDownTitle } from "./SlideDownTitle";
