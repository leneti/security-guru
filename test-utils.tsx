import type { ReactElement, ReactNode } from "react";
import { MantineProvider } from "@mantine/core";
import { render, type RenderOptions } from "@testing-library/react";
import { resolver, theme } from "@site/constants";

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return (
    <MantineProvider theme={theme} cssVariablesResolver={resolver}>
      {children}
    </MantineProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
