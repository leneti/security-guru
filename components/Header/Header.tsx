import {
  createStyles,
  Menu,
  Group,
  clsx,
  UnstyledButton,
  Box,
} from "@mantine/core";
import { useMediaQuery, useWindowScroll } from "@mantine/hooks";
import { IconChevronDown } from "@tabler/icons";
import Link from "next/link";
import { Logo } from "@site/components/Logo";
import { BurgerMenu } from "@site/components/Header/BurgerMenu";
import { menuLinks } from "@site/constants/menu-links";
import { headerHeight } from "@site/constants/measurements";
import { getBGColor } from "@site/utils/getBGColor";
import { useGlobalStyles } from "@site/utils/useGlobalStyles";

const useStyles = createStyles(
  (theme, { isScrolled }: { isScrolled: boolean }) => ({
    outer: {
      width: "100%",
      backgroundColor: getBGColor(theme),
      position: "sticky",
      top: 0,
      zIndex: 2,
      boxShadow: isScrolled ? theme.shadows.md : "none",
      height: headerHeight,
    },

    inner: {
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center",
      height: headerHeight,
      maxWidth: theme.breakpoints.xl,
      marginLeft: "auto",
      marginRight: "auto",

      [`@media (max-width: calc(${theme.spacing.xl} * 2 + ${theme.breakpoints.xl}))`]:
        {
          paddingLeft: `calc(${theme.spacing.xl} - ((100vw - ${theme.breakpoints.xl}) / 2))`,
          paddingRight: `calc(${theme.spacing.xl} - ((100vw - ${theme.breakpoints.xl}) / 2))`,
        },

      [theme.fn.smallerThan("xl")]: {
        paddingLeft: theme.spacing.xl,
        paddingRight: theme.spacing.xl,
      },
    },

    linkGroup: {
      display: "flex",
    },

    link: {
      display: "block",
      lineHeight: 1,
      padding: "8px 12px",
      borderRadius: theme.radius.sm,
      textDecoration: "none",
      color: theme.colors.dark[0],
      fontSize: theme.fontSizes.sm,
      fontWeight: 500,

      "&:hover": {
        backgroundColor: theme.colors.dark[4],
      },
    },

    subLink: {
      textDecoration: "none",
      color: theme.colors.dark[0],
      fontSize: theme.fontSizes.sm,
      fontWeight: 500,
    },

    linkLabel: {
      marginRight: 5,
    },

    switch: {
      backgroundColor: theme.colors.dark[5],
      color: theme.fn.primaryColor(),
      border: "none",
      borderRadius: 8,

      "&:hover": {
        backgroundColor: theme.colors.dark[4],
      },
    },
  })
);

export default function Header() {
  const [{ y }] = useWindowScroll();
  const { classes, theme } = useStyles({ isScrolled: y > 0 });
  const { classes: gClasses } = useGlobalStyles();
  const extraSmallScreen = useMediaQuery(
    theme.fn.smallerThan("xs").substring(7)
  );

  const items = menuLinks.map((link) => {
    if ("links" in link) {
      const menuItems = link.links.map(({ url, label }) => (
        <Link key={url} href={url} className={classes.subLink}>
          <Menu.Item>{label}</Menu.Item>
        </Link>
      ));

      return (
        <Menu key={link.label}>
          <Menu.Target>
            <UnstyledButton className={classes.link}>
              <span className={classes.linkLabel}>{link.label}</span>
              <IconChevronDown
                size={12}
                stroke={1.5}
                aria-label="Toggle menu dropdown"
              />
            </UnstyledButton>
          </Menu.Target>
          <Menu.Dropdown>{menuItems}</Menu.Dropdown>
        </Menu>
      );
    }

    return (
      <Link key={link.label} href={link.url} className={classes.link}>
        {link.label}
      </Link>
    );
  });

  return (
    <Box className={classes.outer}>
      <nav className={classes.inner}>
        <Logo size={extraSmallScreen ? 40 : 50} />
        <BurgerMenu />
        <Group
          spacing={5}
          className={clsx(gClasses.bigDisplay, classes.linkGroup)}
          noWrap
        >
          {items}
        </Group>
      </nav>
    </Box>
  );
}
