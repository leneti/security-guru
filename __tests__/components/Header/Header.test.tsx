import { render, screen } from "@site/test-utils";
import { Header } from "@site/components/Header";

jest.mock("@site/components/Logo");
jest.mock("@mantine/core", () => {
  const orgMantineCore = jest.requireActual("@mantine/core");
  return {
    ...orgMantineCore,
    Group: jest.fn(() => <div>Mock Group</div>),
  };
});

describe("Header", () => {
  it.each([{ scrolled: 0 }, { scrolled: 1 }])(
    "matches snapshot when scrolled $scrolled px",
    ({ scrolled }) => {
      jest
        .spyOn(jest.requireActual("@mantine/hooks"), "useWindowScroll")
        .mockReturnValue([{ y: scrolled }]);

      const { container } = render(<Header />);

      expect(container).toMatchSnapshot();
    },
  );
});
