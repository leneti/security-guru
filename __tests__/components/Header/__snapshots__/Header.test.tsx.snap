// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot when scrolled 0 px 1`] = `
<div>
  <div
    class="mantine-15n1ayh"
  >
    <nav
      class="mantine-15v2uj8"
    >
      <a
        data-testid="logo"
        size="50"
      >
        Mocked Logo
      </a>
      <button
        class="mantine-UnstyledButton-root mantine-Burger-root mantine-hxvumk"
        title="Atidaryti meniu"
        type="button"
      >
        <div
          class="mantine-zpoir7 mantine-Burger-burger"
        />
      </button>
      <div>
        Mock Group
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header matches snapshot when scrolled 1 px 1`] = `
<div>
  <div
    class="mantine-1d79mma"
  >
    <nav
      class="mantine-15v2uj8"
    >
      <a
        data-testid="logo"
        size="50"
      >
        Mocked Logo
      </a>
      <button
        class="mantine-UnstyledButton-root mantine-Burger-root mantine-hxvumk"
        title="Atidaryti meniu"
        type="button"
      >
        <div
          class="mantine-zpoir7 mantine-Burger-burger"
        />
      </button>
      <div>
        Mock Group
      </div>
    </nav>
  </div>
</div>
`;
